### PR TITLE
Enhanced RouteDrawer

### PR DESCRIPTION
**Pull request by RoiEXLab**
This enhances the RouteDrawing method a bit.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/1118)

### CODE REVIEW

**Comment by DanVanAtta** [comment]
Would you mind providing some more context @RoiEXLab ? In-code too we could use some more context.

File: src/games/strategy/triplea/ui/MapRouteDrawer.java [hunk 1]
```diff
@@ -246,7 +246,7 @@ protected static Point getPointOnMap(final Point point, final int xOffset, final
     final int y = point.y;
     final int width = dimension.width;
     final int screenWidth = screenDimension.width;
-    if (x - width > xOffset || (xOffset < 0 && xOffset + screenWidth < x)) {
+    if (x - width > xOffset || (xOffset < - width / 2 && xOffset + screenWidth < x)) {
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/MapRouteDrawer.java]: could be of some help to push this boolean logic to variables or helper methods, allows for something to label 'what' the values represent, and gives us some context as to why we are doing the operations. As is, you somewhat have to reverse engineer what is going on. Ideally we would save on that effort via self-documenting code (failing that, throw some comments in)

**Comment by RoiEXLab** [comment]
@DanVanAtta hmm I think you're right...
Closing this for now.
Since I'm not happy with the RouteDrawer myself, I will refactor it now

**Comment by RoiEXLab** [comment]
Good news @DanVanAtta, I refactored the code and transformed the translate method into it's own object which enables us of much more possibilities.
It's not working perfect, but almost as good as the current solution and tweaking shouldn't be too hard...
I think I can get it working in the next couple days before the release...

**Comment by RoiEXLab** [comment]
@DanVanAtta I'd appreciate it if you could give some early general feedback on the change: https://github.com/triplea-game/triplea/compare/master...RoiEXLab:RouteDrawerEnhancement?expand=1

**Comment by RoiEXLab** [comment]
It's working PERFECTLY now... Reopening...

**Comment by RoiEXLab** [comment]
Orrr not - Closing due to the cluttered diff

**Pull request closed without merging**